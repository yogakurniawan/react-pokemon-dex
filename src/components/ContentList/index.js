import React from 'react';
import PropTypes from 'prop-types';
import { Col, Row } from 'react-flexbox-grid';
import { CircularProgress } from 'material-ui';
import List from 'components/List';

function ContentList({ error, loading, showDetail, component, payload }) {
  if (loading) {
    return <Row center="xs">
      <Col xs={12}>
        <CircularProgress size={50} />
      </Col>
    </Row>;
  }

  if (error) {
    return <div>Something went wrong, please try again!</div>;
  }

  if (payload.length) {
    return (
      <Row center="xs" style={{ margin: '20px 10px' }}>
        <Col xs={12} sm={10} md={10} lg={6}>
          <List items={payload} showDetail={showDetail} />
        </Col>
      </Row>
    );
  }

  return null;
}

ContentList.propTypes = {
  loading: PropTypes.bool,
  error: PropTypes.bool,
  payload: PropTypes.array,
  showDetail: PropTypes.func,
};

export default ContentList;
