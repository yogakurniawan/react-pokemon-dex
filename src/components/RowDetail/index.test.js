import React from 'react';
import injectTapEventPlugin from 'react-tap-event-plugin';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import ContentList from './index';

injectTapEventPlugin();

const props = {
  label: "",
  content: ""
}

describe('ContentList', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <MuiThemeProvider>
        <ContentList {...props} />
      </MuiThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});