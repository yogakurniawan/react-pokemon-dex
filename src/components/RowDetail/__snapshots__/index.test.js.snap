// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentList renders correctly 1`] = `
<div
  className="row start-xs"
  style={
    Object {
      "margin": "10px 0",
    }
  }
>
  <div
    className="col-xs-6 col-sm-6 col-md-4 col-lg-4"
  >
    <label
      style={
        Object {
          "fontWeight": 500,
        }
      }
    >
      
    </label>
  </div>
  <div
    className="col-xs-6 col-sm-6 col-md-8 col-lg-8"
  >
    
  </div>
</div>
`;
